### ISSUE TITLE
Edit titleMaps for individual array form items

### ISSUE THREAD

**Issue by thomask** [json-schema-form/angular-schema-form#338]
I use an array type to let a user add one or multiple "filters". The schema looks like this:

````
...
filters: {
    type: "array",
    items: {
        type: "object",
        properties: {
            id: {
                type: "string"
            },
            comparator: {
                type: "string"
            },
            value: {
                type: "string"
            }
        }
    }
}
`````
And here is the form:

`````
....
{
    key: "filters",
    type: "array",
    style: {
        remove: "btn-danger"
    },
    add: "Add filter",
    title: "Filters",
    items: [
        {
            key: "filters[].id",
            type: "select",
            titleMap: [ ],
            onChange: "filterChanged(form.key,modelValue)"
        },
        {
            key: "filters[].comparator",
            type: "select",
            titleMap: [ ]
        },
        {
            key: "filters[].value",
            type: "select",
            titleMap: [ ]
        }
    ]
}
````
The options for the comparator and the value are dependent on the selected filter (id). I can update the options for those fields *globally* by amending the titleMaps in the form object after a change event. However, I want to update titleMaps for individual items in the array. I did some digging around and it looks to me like this is not possible at the moment. Is that correct?

For me ideally the form object would get automatically updated when a new array item is added, so it follows the exact format of the model object, instead of it being of a "prototype" nature like it is now. Is that something that would be doable for me to amend for myself? Or am I opening a can of worms now :)

**Comment by a-laughlin** [comment]
+1 on this question

**Comment by stramel** [comment]
@thomask So just to clarify, you are wanting to dynamically update the `titleMap` of an item in the array?

**Comment by davidlgj** [comment]
TL;DR; Try using  [dynamic-select](https://github.com/OptimalBPM/angular-schema-form-dynamic-select) and see if that solves your use case.

Hi @thomask, updating a select depending on another select (or similar) seems to be a common pattern, the question has popped up a lot, and the standard answers is to create and add-on for it or use one of the existing ones, like [dynamic-select](https://github.com/OptimalBPM/angular-schema-form-dynamic-select).

With that said, let's get to your actual question, could we do a (different) form definition for each item in the array? And one that is updated automatically without redrawing the entire form? Honestly I don't know :) Not easily though I would guess.

Another thing to think about is if the schema should reflect it? Your schema doesn't validate that if one `id` is selected then only these `comparator` values are valid. This isn't easily described in JSON Schema, but you could do some form of it with `oneOf` I guess. It would be interesting to see if JSON Schema v5 has some way to describe these kinds of dependencies.

**Comment by Anthropic** [comment]
Please re-open if you do not feel the question has been answered sufficiently.
